pallets/kensetsu/src/lib.rs

```diff
@@ -0,0 +1,1449 @@
+                    Self::cover_with_protocol(shortage)?;
+                    // close empty CDP, debt == 0, collateral == 0
+                    Self::delete_cdp(cdp_id)?;
+                    kusd_supply_change = cdp.debt;
+                } else {
+                    // partly covered
+                    Self::update_cdp_debt(cdp_id, shortage)?;
+                    kusd_supply_change = proceeds;
+                }
+            } else {
+                Self::burn_treasury(cdp.debt)?;
+                // CDP debt is covered
+                Self::update_cdp_debt(cdp_id, 0)?;
+                kusd_supply_change = cdp.debt;
+                // There is more KUSD than to cover debt and penalty, leftover goes to cdp.owner
+                let leftover = proceeds
+                    .checked_sub(cdp.debt)
+                    .ok_or(Error::<T>::CDPNotFound)?;
+                assets::Pallet::<T>::transfer_from(
+                    &T::KusdAssetId::get(),
+                    &technical_account_id,
+                    &cdp.owner,
+                    leftover,
+                )?;
+            };
+            Self::decrease_collateral_kusd_supply(&cdp.collateral_asset_id, kusd_supply_change)?;
+            Self::deposit_event(Event::Liquidated {
+                cdp_id,
+                collateral_asset_id: cdp.collateral_asset_id,
+                collateral_amount: collateral_liquidated,
+                proceeds,
+                penalty,
+            });
+
+            Ok(())
+        }
+
+        /// Accrues interest on a Collateralized Debt Position (CDP).
+        ///
+        /// ## Parameters
+        ///
+        /// - `_origin`: The origin of the transaction (unused).
+        /// - `cdp_id`: The ID of the CDP to accrue interest on.
+        #[pallet::call_index(7)]
+        #[pallet::weight(<T as Config>::WeightInfo::accrue())]
+        pub fn accrue(_origin: OriginFor<T>, cdp_id: U256) -> DispatchResult {
+            ensure!(Self::is_accruable(&cdp_id)?, Error::<T>::NoDebt);
+            Self::accrue_internal(cdp_id)?;
+            Ok(())
+        }
+
+        /// Updates the risk parameters for a specific collateral asset.
+        ///
+        /// ## Parameters
+        ///
+        /// - `origin`: The origin of the transaction.
+        /// - `collateral_asset_id`: The identifier of the collateral asset.
+        /// - `new_risk_parameters`: The new risk parameters to be set for the collateral asset.
+        #[pallet::call_index(8)]
+        #[pallet::weight(<T as Config>::WeightInfo::update_collateral_risk_parameters())]
+        pub fn update_collateral_risk_parameters(
+            origin: OriginFor<T>,
+            collateral_asset_id: AssetIdOf<T>,
+            new_risk_parameters: CollateralRiskParameters,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+            Self::ensure_risk_manager(&who)?;
```

Comment:
We had a discussion and came up to a decision that it is a responsibility of a risk management team to set risk parameters. Sudo will not have such right because it takes too much time to react. If requirements are changed, I will add sudo.